### ISSUE TITLE
api.login() Question

### ISSUE THREAD

**Issue by SwedzNet** [TheHolyWaffle/TeamSpeak-3-Java-API#102]
So, as for the first part. I understand that. Simple stuff, but when it gets to the api.login() part, what do I put for the first argument and the second argument? The Query login? Orrrr what?

Thanks! :)

**Comment by SwedzNet** [comment]
Also just another quick question, in the config.setHost(), in the string argument can I put a port as well?

**Comment by Scrumplex** [comment]
```
config.setHost("127.0.0.1");
config.setPort(10011);

---

api.login("server query username", "server query password");
```

**Comment by Scrumplex** [comment]
The setHost thing only works this way

**Comment by SwedzNet** [comment]
I did config.setPort(9316); and it says it's not a valid method..
It's offering me to use config.setQueryPort() though, should I use that?

**Comment by Scrumplex** [comment]
Oh yeah sorry. My fault

**Comment by SwedzNet** [comment]
For some reason, when I created a Query Login on the TS3 server, the bot still doesn't seem to log in.. And every time I click Server Query Login, it just wants me to make a new one..

Very odd..

**Comment by SwedzNet** [comment]
Just a quick question, I'm not sure if you know of Bukkit, but I'm using this to make a Bukkit/Spigot plugin. I just want to make sure this works if not put in a runnable program.

**Comment by Scrumplex** [comment]
It would work if you include the library properly. And don't forget to close the connection onDisable() `query.exit()`

**Comment by SwedzNet** [comment]
I put everything from the /src/ file into my project's /src/, along with putting the jar into my build path.

This is my code:
`@Override
	public void onEnable() {
		this.getServer().getPluginManager().registerEvents(this, this);
		ScoreboardLib.setPluginInstance(this);
		
		log.info("Loading CubedBot...");
		
		final TS3Config config = new TS3Config();
		config.setHost("108.61.118.138");
		config.setQueryPort(9316);

		final TS3Query query = new TS3Query(config);
		query.connect();
		
		final TS3Api api = query.getApi();
		api.login("USERNAME", "PASSWORD");
		api.selectVirtualServerById(1);
		api.setNickname("CubedBot");
		
		log.info("CubedBot loaded.");
	}`
This is my error:
`[11:28:00] [Server thread/INFO]: [Core] Enabling Core v1.0
[11:28:00] [Server thread/INFO]: Loading CubedBot...
[11:28:00] [pool-135-thread-1/INFO]: [2016-06-07 11:28:00.697] [SEVERE] [Connection] Disconnected from TS3 server
[11:28:00] [pool-135-thread-1/INFO]: [2016-06-07 11:28:00.698] [SEVERE] [Connection] Disconnected from TS3 server
[11:28:00] [Server thread/ERROR]: Error occurred while enabling Core v1.0 (Is it up to date?)
com.github.theholywaffle.teamspeak3.api.exception.TS3ConnectionFailedException: An error occurred while connecting to the TeamSpeak3 server
	at com.github.theholywaffle.teamspeak3.QueryIO.<init>(QueryIO.java:71) ~[?:?]
	at com.github.theholywaffle.teamspeak3.TS3Query.connect(TS3Query.java:102) ~[?:?]
	at me.Swedz.core.Main.onEnable(Main.java:47) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:321) ~[spigot-1.8.8.jar:git-Spigot-6abac0c-9935adc]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:340) [spigot-1.8.8.jar:git-Spigot-6abac0c-9935adc]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:405) [spigot-1.8.8.jar:git-Spigot-6abac0c-9935adc]
	at org.bukkit.craftbukkit.v1_8_R3.CraftServer.loadPlugin(CraftServer.java:356) [spigot-1.8.8.jar:git-Spigot-6abac0c-9935adc]
	at org.bukkit.craftbukkit.v1_8_R3.CraftServer.enablePlugins(CraftServer.java:316) [spigot-1.8.8.jar:git-Spigot-6abac0c-9935adc]
	at org.bukkit.craftbukkit.v1_8_R3.CraftServer.reload(CraftServer.java:740) [spigot-1.8.8.jar:git-Spigot-6abac0c-9935adc]
	at org.bukkit.Bukkit.reload(Bukkit.java:534) [spigot-1.8.8.jar:git-Spigot-6abac0c-9935adc]
	at org.bukkit.command.defaults.ReloadCommand.execute(ReloadCommand.java:25) [spigot-1.8.8.jar:git-Spigot-6abac0c-9935adc]
	at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:141) [spigot-1.8.8.jar:git-Spigot-6abac0c-9935adc]
	at org.bukkit.craftbukkit.v1_8_R3.CraftServer.dispatchCommand(CraftServer.java:640) [spigot-1.8.8.jar:git-Spigot-6abac0c-9935adc]
	at org.bukkit.craftbukkit.v1_8_R3.CraftServer.dispatchServerCommand(CraftServer.java:626) [spigot-1.8.8.jar:git-Spigot-6abac0c-9935adc]
	at net.minecraft.server.v1_8_R3.DedicatedServer.aO(DedicatedServer.java:411) [spigot-1.8.8.jar:git-Spigot-6abac0c-9935adc]
	at net.minecraft.server.v1_8_R3.DedicatedServer.B(DedicatedServer.java:375) [spigot-1.8.8.jar:git-Spigot-6abac0c-9935adc]
	at net.minecraft.server.v1_8_R3.MinecraftServer.A(MinecraftServer.java:653) [spigot-1.8.8.jar:git-Spigot-6abac0c-9935adc]
	at net.minecraft.server.v1_8_R3.MinecraftServer.run(MinecraftServer.java:556) [spigot-1.8.8.jar:git-Spigot-6abac0c-9935adc]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_91]
Caused by: java.net.ConnectException: Connection refused
	at java.net.PlainSocketImpl.socketConnect(Native Method) ~[?:1.8.0_91]
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350) ~[?:1.8.0_91]
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206) ~[?:1.8.0_91]
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188) ~[?:1.8.0_91]
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392) ~[?:1.8.0_91]
	at java.net.Socket.connect(Socket.java:589) ~[?:1.8.0_91]
	at java.net.Socket.connect(Socket.java:538) ~[?:1.8.0_91]
	at java.net.Socket.<init>(Socket.java:434) ~[?:1.8.0_91]
	at java.net.Socket.<init>(Socket.java:211) ~[?:1.8.0_91]
	at com.github.theholywaffle.teamspeak3.QueryIO.<init>(QueryIO.java:57) ~[?:?]
	... 18 more
[11:28:00] [pool-135-thread-1/INFO]: [2016-06-07 11:28:00.700] [SEVERE] ConnectionHandler threw exception in disconnect handler
java.lang.NullPointerException
	at com.github.theholywaffle.teamspeak3.TS3Query.doCommand(TS3Query.java:161)
	at com.github.theholywaffle.teamspeak3.TS3Query.exit(TS3Query.java:124)
	at com.github.theholywaffle.teamspeak3.api.reconnect.DisconnectingConnectionHandler.onDisconnect(DisconnectingConnectionHandler.java:49)
	at com.github.theholywaffle.teamspeak3.TS3Query$2.run(TS3Query.java:207)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

[11:28:00] [pool-135-thread-1/INFO]: [2016-06-07 11:28:00.701] [SEVERE] ConnectionHandler threw exception in disconnect handler
java.lang.NullPointerException
	at com.github.theholywaffle.teamspeak3.TS3Query.doCommand(TS3Query.java:161)
	at com.github.theholywaffle.teamspeak3.TS3Query.exit(TS3Query.java:124)
	at com.github.theholywaffle.teamspeak3.api.reconnect.DisconnectingConnectionHandler.onDisconnect(DisconnectingConnectionHandler.java:49)
	at com.github.theholywaffle.teamspeak3.TS3Query$2.run(TS3Query.java:207)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)`

**Comment by Firedroide** [comment]
Closed, seems to be continued in #103.
